ledger/eval/eval.go

```diff
@@ -1461,6 +1430,43 @@ func (eval *BlockEvaluator) endOfBlock() error {
 		}
 	}
 
+	// Try to pay the proposer.
+	{
+		proposer := eval.block.Proposer()
+		payout := eval.block.ProposerPayout()
+		// The proposer won't be present yet when generating a block
+		if !proposer.IsZero() {
+			// We don't propagate the error here, we simply declare that an
+			// illegal payout is not made.  This protects us from stalling if
+			// there is ever an error in which the generation code thinks the
+			// payout will be legal, but it turns out not to be.  That would be
+			// a programming error in algod, but not worth stalling over.
+			if !payout.IsZero() {
+				err2 := eval.state.Move(eval.block.FeeSink, proposer, payout, nil, nil)
+				if err2 != nil {
+					logging.Base().Warnf("Unable to payout %d to %v: %s",
```

Comment:
I was special casing the Move because I wasn't properly handling MinBalance. It was a stopgap to the real solution which appears now.  We no longer pay incentives to closed accounts (that seems like a simpler rule than paying if it brings them over min balance). So these errors should never occur, but if they do, we propogate.

This should mean that if there is a problem I haven't forseen, other nodes will reject the block, and some other proposer will perhaps be able to propose.